gcpim.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	imtypes "cloud-android-orchestration/api/instancemanager/v1"
+	compute "cloud.google.com/go/compute/apiv1"
+	"context"
+	"errors"
+	"github.com/google/uuid"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	// TODO(b/220891296): Make this configurable
+	projectId   = "google.com:cloud-android-jemoreira"
+	sourceImage = "projects/cloud-android-releases/global/images/cuttlefish-google-vsoc-0-9-21"
+	networkName = "projects/cloud-android-jemoreira/global/networks/default"
+)
+
+const (
+	namePrefix  = "cf-"
+	labelPrefix = "cf-"
+)
+
+var newUUIDString = func() string {
```

Comment:
This should be a function instead of a variable

Reply:
we can do much better than having a non constant global variable. One option is to receive a uuidFactory of the same type as your variable as a parameter in the constructor, or even better to have it be an instance variable created with a default value in the constructor but replaceable through a setter function.

Reply:
this variable is only accessible within its package, which is how white-box testing is done in Golang, then the argument is about white-box testing vs black-box testing which is what you suggest. Now, context is key here to decide which way to go. Should we make passing a uuidFactory part of the public API and add the extra work that comes with it or can we keep it internally and do white-box testing? In this case specifically, I think passing and handling a uuidFactory is dirtying the public API as UUID generation could be kept internally. If having another use case different that testing to move it to the public API I won't argue against it.

I'm aware, this code uses the main package right now, I tried to have it on its own package but right now is impossible without moving other code likes errors.go and types.go into their own packages different than main, otherwise we have cyclic dependencies. I'd like the change of splitting main into different packages to be done in a different PR to keep this one focused on the creation process only.

Reply:
You don't have to make it part of the public API, you can create a setter in the class that is not exported from the package the same way you're doing with the global variable.

With regards to white box vs black box testing: I prefer to add something to the public API to enable testing than to have my object access a mutable global variable. Mutable global variable are a known source of hard to debug bugs while an extra element in the API will only confuse the users a little and force them to read documentation.

Reply:
due the current lack of constraints on modifying the package level variable, we decided to move the uuid factory to the public api.